docs/remote/create-dev-container.md

```diff
@@ -56,253 +43,162 @@ All of the predefined container configurations you can pick from come from the [
 
-```yaml
-# Change user for VS Code and sub-processes (terminals, tasks, debugging)
-"remoteUser": "your-user-name-here",
-# Or change the user for all container processes
-"containerUser": "your-user-name-here"
+```bash
+# If sudo is installed and configured
+sudo apt-get update
+sudo apt-get install <package>
 ```
 
-You can also add **additional local mount points** to give your container access to other locations using the `mounts` property.
-
-For example, you can mount your home / user profile folder:
+Let's say you want to install Git. You could run the following commands in the integrated terminal in VS Code:
 
-```json
-"mounts": [
-    "source=${localEnv:HOME}${localEnv:USERPROFILE},target=/host-home-folder,type=bind,consistency=cached"
-]
+```bash
+# If sudo is installed and configured
+sudo apt-get update
+# Install Git
+sudo apt-get install git
 ```
 
-You can also reference `"${localWorkspaceFolder}` if you need to mount something from the local filesystem into the container.
-
-The `runArgs` property supports the same list of arguments as the [docker run](https://docs.docker.com/engine/reference/commandline/run/) command and can be useful for a wide variety of scenarios including [setting environment variables](/docs/remote/containers-advanced.md#adding-environment-variables).
+### Rebuilding
 
-If your application was built using C++, Go, or Rust, or another language that uses a **ptrace-based debugger**, the `runArgs` property can also be used to configure needed runtime security and capability settings.
+When editing the contents of the `.devcontainer` folder, you'll need to rebuild for changes to take effect. Use the **Remote-Containers: Rebuild Container** command for your container to update.
 
-For example:
+However, if you **rebuild** the container, you will have to **reinstall** anything you've installed manually. To avoid this problem, you can use a series of commands in the `postCreateCommand` property in `devcontainer.json`:
 
 ```json
 {
-    "name": "My Go App",
-    "dockerFile": "Dockerfile",
+    "image": "mcr.microsoft.com/vscode/devcontainers/typescript-node:0-12",
     "extensions": [
-        "golang.go"
+        "dbaeumer.vscode-eslint"
     ],
-    "runArgs": [
-        "--cap-add=SYS_PTRACE",
-        "--security-opt",
-        "seccomp=unconfined"
-    ]
+    "forwardPorts": [ 3000 ],
+    "postCreateCommand": "sudo apt-get update && sudo apt-get install git",
 }
 ```
 
-While less efficient than a custom Dockerfile, you can also use the `postCreateCommand` property to **install additional software** that **may not be in your base image** or for cases where you would prefer **not to modify a deployment Dockerfile** you are reusing.
+The `postCreateCommand` is run once the container is running, so you can also use the property to run commands like `npm install` or to execute a shell script in your source tree (if you have mounted it).
```

Comment:
In reflecting back, two tips I'm wondering if we should add here is that:

1. There's also a postStartCommand which executes every time
2. The command needs to exit or the container won't start.

I've seen bugs logged and we saw Dina hit really both of these by adding an application start to postCreateCommand.

Reply:
That's a good idea. I feel like it'd make sense to include those tips- I added some information about them here, let me know what you think.